Framework/PythonInterface/mantid/geometry/src/Exports/Instrument.cpp

```diff
@@ -74,6 +75,12 @@ void export_Instrument() {
            "Return the valid to :class:`~mantid.kernel.DateAndTime` of the "
            "instrument")
 
+      .def("getFilename", &Instrument::getFilename, arg("self"), return_value_policy<copy_const_reference>(),
+           "Return the name of the file that the original IDF was from")
+
+      .def("setFilename", &Instrument::setFilename, (arg("self"), arg("filename")),
+           "Return the name of the file that the original IDF was from")
```

Comment:
Looks like this doc string for setter is duplicate of getter.

Just out of interest, does setting a different filename actually re-load the IDF (like calling `LoadInstrument`) - it doesn't look like it does here (just updates `m_filename`)
https://github.com/mantidproject/mantid/blob/789921cf7a96883cd6d3a126277f99af5742ee3d/Framework/Geometry/src/Instrument.cpp#L845-L850
so could exposing this to python could lead to problems? Depends on what you want to use this for I guess!

Reply:
The SNAP team is overwriting the instrument file after doing a subsampling to make the instrument effectively have giant pixels.

Reply:
So is this grouping neighbouring pixels or similar? Not sure what the benefit of having a separate IDF is (when you group detectors, e.g. when focussing it just keeps track of the detector IDs contributing to each spectrum), but obviously I'm not familiar with SNAP stuff! My one query would be does it make sense to be able to set the filename when it could be out of date with the IDF loaded (assuming `LoadInstrument` set this `m_filename`)?